Sources/SymbolKit/SymbolGraph/Relationship/Relationship.swift

```diff
@@ -107,6 +103,93 @@ extension SymbolGraph {
     }
 }
 
+extension SymbolGraph.Relationship {
+    struct CodingKeys: CodingKey, Hashable {
+        let stringValue: String
+        let encoder: ((Self, Mixin, inout KeyedEncodingContainer<CodingKeys>) throws -> ())?
+        let decoder: ((Self, KeyedDecodingContainer<CodingKeys>) throws -> Mixin?)?
```

Comment:
Where would you like them to be stored? I can only build the `encoder`/`decoder` closures from inside an extension to `Mixin`, otherwise I cannot access the `Self` I need for calling `container.decode`/`container.encode`.

 >  I can see why you've made these closures for type-erasure purposes, but could we instead just keep a mapping from keys to the types they encode to/decode from, such as a [String: Mixin.Type] dictionary? Then, we'd look up the mixin type for a given key and use that for decoding.

I don't think that's possible with our target language version (might be possible from Swift 5.6):

<img width="708" alt="image" src="https://user-images.githubusercontent.com/21169289/180613109-dac030ef-4e1d-4087-86be-f3ba549b1f53.png">

Reply:
This would definitely be possible. I'd prefer to make a clean cut, though. If we want to go this route, I'd prefer to bump the language version for SymbolKit (and all of apple/swift-docc) to 5.7 and make `Mixin` `Hashable` and `Codable` in one move. I think it would be really weird if bumping the language version to 5.7 in a project using SymbolKit would cause it to behave entirely different.